geomfn: ensure NaN never intersects/covers/within

otan opened the pull request:
Resolves #79992

Release note (bug fix): Fixed a bug where NaN coordinates when using
ST_Intersects/ST_Within/ST_Covers would return true instead of false for
point in polygon operations.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/80136)
<!-- Reviewable:end -->

otan committed:
geomfn: ensure NaN never intersects/covers/within

Release note (bug fix): Fixed a bug where NaN coordinates when using
ST_Intersects/ST_Within/ST_Covers would return true instead of false for
point in polygon operations.

rytaft reviewed (commented):
Does this fix the test case in #79992? Or is it possible that tlp will continue to find this as an issue?

If it does fix it, can you backport the fix to 22.1 and 21.2? Thanks!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/80136#-:-N00njy8EnXyr0W90qPu:bpsx28i)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss)



<!-- Sent from Reviewable.io -->

rafiss commented on pkg/geo/geomfn/topology_operations.go:
```diff
@@ -208,17 +208,36 @@ func MinimumRotatedRectangle(g geo.Geometry) (geo.Geometry, error) {
+		return false
+	}
+	// Don't use `:= range []float64{...}` to avoid memory allocation.
+	isNaN := func(ord float64) bool {
```
very advanced helper function :D

otan commented:
> If it does fix it, can you backport the fix to 22.1 and 21.2? Thanks!

it does, but i need to backport two PRs so that i don't get a merge conflict so i'll do it by hand when this merges

bors r=rafiss

otan commented:
oh wait that commit that may merge conflict is already merged, blathers should do it
